src/hip/hip_build_utils.cpp

```diff
@@ -73,7 +73,7 @@ static boost::filesystem::path HipBuildImpl(boost::optional<TmpDir>& tmp_dir,
     auto env = std::string("");
 
     if(params.find("-std=") == std::string::npos)
-        params += " --std=c++11";
+        params += " --std=c++17";
```

Comment:
This affects all the HIP kernels included into the library (that do not explicitly specify C++ standard) and therefore requires extensive testing (and should be done in a separate PR at least). Please revert this and produce the necessary option in the naive solvers.